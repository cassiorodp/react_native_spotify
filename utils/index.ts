import * as SecureStore from 'expo-secure-store';

const secureStore = async (key: string, value: string) => {
  await SecureStore.setItemAsync(key, value);
};

const getSecureStore = (key: string) => {
  return SecureStore.getItem(key);
};

export { secureStore, getSecureStore };
