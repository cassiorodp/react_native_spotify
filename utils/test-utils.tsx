import {
  RenderAPI,
  RenderOptions,
  render,
} from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from 'react-query';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  options?: RenderOptions
): RenderAPI => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
