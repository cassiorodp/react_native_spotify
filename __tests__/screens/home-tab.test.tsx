import Index from '@/app/tabs/home';
import { render } from '@/utils/test-utils';

describe('Home tab component', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<Index />);

    expect(toJSON()).toMatchSnapshot();
  });
});
