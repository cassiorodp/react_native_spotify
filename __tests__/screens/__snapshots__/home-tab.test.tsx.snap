// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tab component should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      undefined,
      {
        "flex": 1,
        "paddingHorizontal": 30,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "rgb(28, 28, 30)",
          },
          {
            "fontFamily": "Rubik-Regular",
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Top artistas
    </Text>
    <ActivityIndicator
      color="rgb(0, 122, 255)"
      size="small"
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginTop": 30,
      }
    }
  >
    <ActivityIndicator
      color="rgb(0, 122, 255)"
      size="large"
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
</RNCSafeAreaView>
`;
